### PR TITLE
[templates] Add script to restore yarn.lock

### PR DESCRIPTION
**Pull request by sc-addypathania**
When following our new workflow (`npm run watch`) the root yarn.lock file will result in dependencies being added for the initialized app (since we are using yarn workspace). This is a problem because developers must be aware of these changes and not commit them to source control. This script will restore the `yarn.lock` file after app initialization is complete.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/scripts/watch-templates.ts [hunk 1]
```diff
@@ -2,18 +2,27 @@ import chalk from 'chalk';
...
 import { initRunner } from '../src/init-runner';
+import { postInstall } from './post-install';
 
 chokidar
   .watch(path.join(process.cwd(), '.\\src\\templates'), { ignoreInitial: true })
   .on('ready', () => ready())
   .on('all', (event, path) => callback(event, path));
 
+/**
+ *
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/scripts/watch-templates.ts:13]: Can you fill description?

File: packages/create-sitecore-jss/scripts/watch-templates.ts [hunk 2]
```diff
@@ -2,18 +2,27 @@ import chalk from 'chalk';
...
+ */
 async function ready() {
   console.log(chalk.green('Initializing app...'));
   await initializeApps(false);
+  postInstall();
   console.log(chalk.green('Initializing app complete. Watching for changes...'));
 }
 
+/**
+ * @param event
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/scripts/watch-templates.ts:23]: can you describe params and function?

File: packages/create-sitecore-jss/package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "build": "npm run clean && tsc && ts-node ./scripts/build-templates.ts",
     "clean": "del-cli dist types",
     "lint": "eslint --no-eslintrc -c .eslintrc ./src/**/*.ts",
-    "watch": "ts-node ./scripts/watch-templates.ts",
+    "watch": "ts-node ./scripts/watch-templates.ts && ./scripts/post-install.js",
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/package.json:10]: `watch-templates` doesn't exit from the process, as I remember, did you test and check that `post-install` will be executed?

File: packages/create-sitecore-jss/scripts/post-install.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿const execSync = require('child_process').execSync;
+
+// restore yarn.lock file post initializing a sample app using watch script.
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/scripts/post-install.ts:3]: Can you describe, the reason why we restore this file?

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: packages/create-sitecore-jss/package.json (hunk 3 above)
Comment (sc-addypathania) [reply packages/create-sitecore-jss/package.json:10]: @illiakovalenko yes I have tested the changes, post-install gets executed and the watch script doesn't exit.

File: packages/create-sitecore-jss/scripts/post-install.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿const execSync = require('child_process').execSync;
+
+// restore dependencies added to yarn.lock file post initializing a sample app using the watch script.
```
Comment (ambrauer) [inline packages/create-sitecore-jss/scripts/post-install.ts:3]: This works if you are _only developing templates_, but creates a bit of a reverse issue when you are _also developing packages_. In this case, you may revert legitimate yarn.lock changes (e.g. you `yarn add` a new dependency to `sitecore-jss` package). 

Maybe we make this another `watch.json` config option to allow you to toggle it on/off depending on what you're working on?

Comment (sc-addypathania) [reply packages/create-sitecore-jss/scripts/post-install.ts]: Are we talking about adding a dependency while initializing an app using a watch script?

Comment (ambrauer) [reply packages/create-sitecore-jss/scripts/post-install.ts]: I was thinking (1) adding a dependency (working on base package), and then (2) you go to work on something in an initializer and do a yarn watch. Your legit yarn lock change from 1 would be wiped out.

File: packages/create-sitecore-jss/watch.json.example [hunk 6]
```diff
@@ -12,6 +12,9 @@
     "language": "da-DK",
     "fetchWith": "GraphQL",
     "prerender": "SSG",
-    "hostName": "sitecore-jss-nextjs.dev.local"
+    "hostName": "sitecore-jss-nextjs.dev.local",
+    // if you're developing packages i.e install dependencies before initializing an app.
+    // Set devTemplates to false or else your yarn.lock will get restored.
+    "devTemplates": true
```
Comment (ambrauer) [inline packages/create-sitecore-jss/watch.json.example:18]: I think we should tie this config name directly to the action it's controlling. "restoreLockfile"?

File: packages/create-sitecore-jss/scripts/watch-templates.ts [hunk 7]
```diff
@@ -2,30 +2,53 @@ import chalk from 'chalk';
...
   const color = event === 'add' ? chalk.green : event === 'unlink' ? chalk.red : chalk.white;
   console.table(color(`${event} ${path}`));
   await initializeApps(true);
 }
 
+/**
+ * restore dependencies added to yarn.lock file post initializing a sample app using the watch script.
+ * this is necessary so that these dependencies dont get committed to the source control.
+ */
+function postInstall() {
```
Comment (ambrauer) [inline packages/create-sitecore-jss/scripts/watch-templates.ts:36]: Name this to match the action and config (see other comment). "restoreLockfile()"?

**Review by ambrauer — APPROVED** [review]
Looks good, just a couple naming suggestions

**Pull request merged**